### PR TITLE
docs: improve the project readme

### PR DESCRIPTION
**Pull request by SoftwareGuy**
This PR improves the project's readme file, by stating what Mirage is and what it will provide should one use it for their networked project. It also encourages people to get involved and help diagnose and fix issues that may occur with Mirage.

Some paragraphs may need adjustments and I am completely open to discussion about it - please feel free to disagree and provide better wording or better way of expressing the dot point.

It also acknowledges Mirror as a _distant sister_ and tries to keep things cool (no jabs or anything). I originally consulted with James regarding this but think of this as just improving the sales pitch that puts Mirage on the networking map.

**Original commit message below.**

* Rework intro section (now "What is Mirage?")
* Update intro section to say Mirage is a high performance multi-genre networking stack, add in that we welcome feature requests/ideas, we review PRs regularly and welcome improvements and bug fixes
* Cleanup Transport and Sockets
* Recommend Unity 2020 LTS since readme originally said 2020.1 or later
* TLC, polish and proof-reading

### CODE REVIEW

**Comment by SoftwareGuy** [comment]
I swear that pr name linter hates me...

**Comment by James-Frowen** [comment]
> I swear that pr name linter hates me...

needs to start with lowercase, also should be `docs: ` for this one :D

File: README.md [hunk 1]
```diff
@@ -10,99 +10,102 @@
...
 * Components can be added in child objects
-* Your client can connect to multiple servers. For example chat server and game server
-* Modular,  use only the components you need.
+* Modular API: You only use the components you need.
 * Error handling
 * [Version defines](https://docs.unity3d.com/Manual/ScriptCompilationAssemblyDefinitionFiles.html#define-symbols)
 * Server Rpcs can [return values](https://miragenet.github.io/Mirage/Articles/Guides/RemoteCalls/ServerRpc.html)
-* The default transport features DoS prevention
-* The default transport has CRC64 integrity check
+* A built-in reliable UDP default transport featuring DoS prevention and CRC64 integrity checks
```
Comment (James-Frowen) [inline README.md:70]: I'm not sure the "featuring DoS prevention and CRC64 integrity checks" is in the new transport, we should probably remove this part for now.

And make we can open a separate github issue if people want the CRC64 integrity stuff

**Comment by James-Frowen** [comment]
Looks good, I like the first paragraph a lot more than the old one.

File: README.md (hunk 1 above)
Comment (SoftwareGuy) [reply README.md:70]: Alright, will remove it - I believe that was a fragment from the old readme.

File: README.md [hunk 2]
```diff
@@ -10,99 +10,102 @@
...
+Mirage has some notable differences from its distant sister, Mirror. The table below briefly details them:
 
 | Mirage                                              | Mirror                                 |
 | --------------------------------------------------- | -------------------------------------- |
-| Install via Unity Package Manager                   | Install from Asset Store               |
+| Installs via Unity Package Manager                  | Installs from Asset Store              |
 | Errors are thrown as exceptions                     | Errors are logged                      |
 | `[ServerRpc]`                                       | `[Command]`                            |
-| `[ClientRpc(target=Client.Owner)]`                  | `[TargetRpc]`                          |
+| `[ClientRpc(target = Client.Owner)]`                | `[TargetRpc]`                          |
```
Comment (phodoval) [inline README.md:51]: I believe the `Client.Owner` has been renamed to `RpcTarget.Owner`. Also maybe adding a link to our migration guide docs would be nice.

**Review by phodoval — CHANGES_REQUESTED** [review]
One small tweak, otherwise looks good. :)

File: README.md (hunk 2 above)
Comment (SoftwareGuy) [reply README.md]: Adjusted in the latest commit.

File: README.md (hunk 1 above)
Comment (SoftwareGuy) [reply README.md]: Removed in the latest commit.

**Review by phodoval — APPROVED** [review]
LGTM

**Pull request merged**